### PR TITLE
Add Grad-CAM++ functionality

### PR DESCRIPTION
**Pull request by farazali7**
This PR is for adding Grad-CAM++ functionality into the explainability part of the repo. Full details about Grad-CAM++ can be found in the [paper.](https://arxiv.org/pdf/1710.11063.pdf)

Summary of changes:

- Added `EXPLAINABILITY >> GRAD_CAM` section to `config.yml` for specifying whether to compute heatmaps with normal Grad-CAM or Grad-CAM++
- Added function for computing Grad-CAM++ heatmap using formula from paper
- Minor refactoring of `gradcam.py` to promote code reuse

### CODE REVIEW

File: src/explainability/gradcam.py [hunk 1]
```diff
@@ -1,119 +1,170 @@
...
+
+        last_conv_sum = tf.reduce_sum(last_conv_layer, axis=(0, 1, 2))
+
+        alpha_num = second_grads
+        alpha_denom = second_grads * 2.0 + third_grads * last_conv_sum
+        alpha_denom = np.where(alpha_denom != 0.0, alpha_denom, EPSILON)
+
+        alphas = alpha_num / alpha_denom
+        alpha_norm = np.sum(alphas, axis=(0, 1))
+        alphas /= alpha_norm
```
Comment (bvanberl) [inline src/explainability/gradcam.py:94]: Where in the paper do they normalize the alphas?

Comment (farazali7) [reply src/explainability/gradcam.py:94]: Alpha normalization was not stated in the paper. However, the author's perform normalization in their actual implementation found on github, so we did the same. I'm not too sure why, but one theory could be to bring the magnitude of the alphas closer to the first partial derivative they get multiplied with.

File: src/explainability/gradcam.py [hunk 2]
```diff
@@ -1,119 +1,170 @@
...
+
+    @staticmethod
+    def get_gradcam_plusplus_heatmap(model, img):
+        '''
+        Compute the raw heatmap for the given image using the Grad-CAM++ formula for linearization weights.
+        :param model: Model to process given image through
+        :param img: Preprocessed image in dataset
+        :return: Unactivated, unnormalized heatmap
+        '''
+        # Obtain first, second, and third gradients of output with respect to last convolutional layer weights
```
[the remark this answers is not in the document]
Comment (farazali7) [reply src/explainability/gradcam.py:76]: Renamed in latest commit.
